var/spack/repos/builtin/packages/ccache/package.py

```diff
@@ -0,0 +1,45 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Ccache(AutotoolsPackage):
+    """ccache is a compiler cache. It speeds up recompilation by caching
+    previous compilations and detecting when the same compilation is being done
+    again."""
+
+    homepage = "https://ccache.samba.org/"
+    url      = "https://www.samba.org/ftp/ccache/ccache-3.3.3.tar.gz"
+
+    version('3.3.3', 'ea1f95303749b9ac136c617d1b333eef')
+    version('3.3.2', 'b966d56603e1fad2bac22930e5f01830')
+    version('3.3.1', '7102ef024cff09d353b3f4c48379b40b')
+    version('3.3'  , 'b7ac8fdd556f93831618483325fbb1ef')
+    version('3.2.9', '8f3f6e15e75a0e6020166927d41bd0b3')
+
+    depends_on('autoconf', type='build')
```

Comment:
I suppose another way for me to phrase what I said previously is to ask whether there are any specific resources/guides on how to best put together an `autotools` based package? The [packaging guide](https://spack.readthedocs.io/en/latest/packaging_guide.html#packaging-guide) and [tutorial](https://spack.readthedocs.io/en/latest/tutorial_sc16_packaging.html?highlight=autotools) don't offer any guidance other than a brute-force approach where you start with nothing, incrementally build, and add whatever it takes to satisfy any build errors. 

Assuming one reads the project's documentation to find the dependent libraries _a priori_, what is the most practical way to determine if a project will need one of these?
```
# depends_on('autoconf', type='build')
# depends_on('automake', type='build')
# depends_on('libtool', type='build')
```